src/IssueViz.Security/HotspotsList/TableDataSource/HotspotTableEntryWpfElementFactory.cs

```diff
@@ -18,34 +18,72 @@
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
-using System;
 using System.Drawing;
 using System.Windows;
 using System.Windows.Controls;
 using System.Windows.Data;
 using System.Windows.Media;
+using Microsoft.VisualStudio.Imaging;
 using Microsoft.VisualStudio.Shell.Interop;
+using SonarLint.VisualStudio.IssueVisualization.Models;
 using Color = System.Windows.Media.Color;
 
 namespace SonarLint.VisualStudio.IssueVisualization.Security.HotspotsList.TableDataSource
 {
-    internal interface INonNavigableFrameworkElementFactory
+    internal interface IHotspotTableEntryWpfElementFactory
     {
         FrameworkElement Create(string content);
     }
 
-    internal class NonNavigableFrameworkElementFactory : INonNavigableFrameworkElementFactory
+    internal class HotspotTableEntryWpfElementFactory : IHotspotTableEntryWpfElementFactory
     {
         private readonly IVsUIShell2 vsUiShell;
+        private readonly IAnalysisIssueVisualization issueVisualization;
 
-        public NonNavigableFrameworkElementFactory(IServiceProvider serviceProvider)
+        public HotspotTableEntryWpfElementFactory(IVsUIShell2 vsUiShell, IAnalysisIssueVisualization issueVisualization)
         {
-            vsUiShell = serviceProvider.GetService(typeof(SVsUIShell)) as IVsUIShell2;
+            this.vsUiShell = vsUiShell;
+            this.issueVisualization = issueVisualization;
         }
 
         public FrameworkElement Create(string content)
         {
-            var control = new TextBlock
+            var control = new Border
+            {
+                ContextMenu = CreateContextMenu(),
```

Comment:
This probably needs some explanation: we can't create a context menu for the row or the table as a whole, so we have to create the same context menu for each column element in the row?